### ISSUE TITLE
Array reference+hash reference?

### ISSUE THREAD

**Issue by benkasminbullock** [simonecesano/CSS-Parser-Regexp#1]
When I tried the synopsis example, the second print gives
````
$VAR1 = {
          '.someclass' => [
                            {
                              'color' => 'limegreen',
                              'font-size' => '12pt'
                            }
                          ]
        };
````
I tried it on a few other things and always there is [{ }] around all the rules, I think the [] is unnecessary isn't it?

**Comment by simonecesano** [comment]
I have a hunch it might come in handy later - that's because the same selector could be specified in different places. As an alternative I would have to merge the hashes as they come up. The arrayref lets me keep track of what parsing scooped up and merge them later (it makes even more sense as I look at the other [issue](https://github.com/simonecesano/CSS-Parser-Regexp/issues/2) you raised)

**Comment by benkasminbullock** [comment]
I don't think that the order of rules is significant for a particular thing but I'm not sure about that. If they were significant then the hash reference would be wrong. If you always get an array reference with only one element, which is a hash reference, it's hard to see how it could possibly be useful, but it's your decision of course. I won't close it but please close this issue if you want to.

**Comment by simonecesano** [comment]
Thanks - I looked into this. No - it's not always only one hashref in an arrayref. When the same selector pops up twice at different spots it will result in more than one hashref. It happens for example with [style_02.css](https://github.com/simonecesano/CSS-Parser-Regexp/blob/main/css-tests/style_02.css). However, I realized that it causes issues when doing the roundtrip parse => stringify => parse again: this results in two different tree structures. I'll need to make up my mind if that's ok. Also - the hashref vs. arrayref rule is used to distinguish between selector paths and styles - see [here](https://github.com/simonecesano/CSS-Parser-Regexp/blob/main/lib/CSS/Parser/Regexp.pm#L351)

**Comment by benkasminbullock** [comment]
With this example code,
````
use CSS::Parser::Regexp;
use Data::Dumper;

$\ = "\n";

my $css = <<CSS
  :root {
    --red-color: red;
  }
  .red-box {
    background:red;
    background: var(--red-color);
    height: 200px;
    width:200px;
  }
CSS
;

my $p = CSS::Parser::Regexp->new;

$p->parse($css);

print Dumper($p->to_tree);
````
it prints
````
VAR1 = {
          ':root' => [
                       {
                         '--red-color' => 'red'
                       }
                     ],
          '.red-box' => [
                          {
                            'background' => 'var(--red-color)',
                            'height' => '200px',
                            'width' => '200px'
                          }
                        ]
        };

````
so the first background seems to be overwritten.
